pkg/sfu/buffer/dependencydescriptorparser.go

```diff
@@ -0,0 +1,104 @@
+package buffer
+
+import (
+	"fmt"
+
+	dd "github.com/livekit/livekit-server/pkg/sfu/dependencydescriptor"
+
+	"github.com/livekit/protocol/logger"
+)
+
+type videoLayer struct {
+	spatial  int
+	temporal int
+}
+
+type DependencyDescriptorParser struct {
+	structure         *dd.FrameDependencyStructure
+	ddExt             uint8
+	logger            logger.Logger
+	onMaxLayerChanged func(int, int)
+	decodeTargetLayer []videoLayer
+}
+
+func NewDependencyDescriptorParser(ddExt uint8, logger logger.Logger, onMaxLayerChanged func(int, int)) *DependencyDescriptorParser {
+	logger.Infow("creating video stream receiver", "ddExt", ddExt)
+	return &DependencyDescriptorParser{
+		ddExt:             ddExt,
+		logger:            logger,
+		onMaxLayerChanged: onMaxLayerChanged,
+	}
+}
+
+func (r *DependencyDescriptorParser) Parse(ep *ExtPacket) error {
```

Comment:
Would be good for this to take just an RTP packet and return parsed values. `ExtPacket` is okay, but it has a lot of other stuff and it would be good to keep standalone modules like not have that dependency. This is also modifying ExtPacket which would be nice to avoid.